### PR TITLE
Featured blog posts

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/4624 by introducing "featured blog pages" for the blog index page, with template updates to make sure they show up with the same styling as the "featured posts"/"latest news" on the homepage

- /blog should show 2 featured posts, with those not showing the subsequent set of blog posts
- /category pages should not show featured posts
- /tags pages technically shouldn't either, but we don't really expose those so... win?

Test URL: https://foundation-s-featured-b-oyndlu.herokuapp.com/en/blog/

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_page.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "./index_page.html" %}
+
+{% load wagtailcore_tags wagtailimages_tags i18n %}
+
+{% block featured %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_page.html:5]: the new part, which injected featured posts as long as we're not on a /category or /tag view

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_page.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  {% if not filtered %}
+    <div class="row">
+      {%  for feature in page.specific.featured_pages.all %}
+      <div class="col-6 mb-5">
+        {% include "./fragments/featured_blog.html" with blog=feature %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blog_index_page.html:10]: refactored-to-stand-alone featured post as found on the homepage

**Comment by Pomax** [comment]
@kristinashu using the standard template for latest news/featured blog as we do on the homepage can lead to what feels like oddly misaligned hbars:

image

We don't notice this on the homepage because we only feature one thing per section, but do we want to some kind of "height based on lines" with a `text-overflow: ellipsis` if the SEO search text runs longer than that, so that the hbars are guaranteed height-aligned?

**Comment by kristinashu** [comment]
Thank you @Pomax! 

## hbar alignment 
The misaligned hbars issue seems to have been fix on our homepage's "Featured stories" section. Can you you do something like that? (screen cap from prod)
image


## Narrow widths
The two posts will need to stack on top of each other narrower widths. (see issue in this screen cap from PR)
image

Could it stack similar to the homepage? (screen cap from prod)
Screen Shot 2020-05-25 at 4 44 09 PM


## Selection order
For selection which posts to feature, can the selection order show the most recent at the top? Related to https://github.com/mozilla/foundation.mozilla.org/issues/3714.

**Comment by Pomax** [comment]
The entries can be ordered in the CMS itself, so it feels like that doesn't need explicit hardcoding:

image

**Comment by kristinashu** [comment]
That feature is perfect. What I mean when staff choose which blog post to feature there. Can the most recent blog post show up at the top? I think you updated this on the homepage recently. 
(screen cap from PR)
image

**Comment by kristinashu** [comment]
Or actually, maybe it is showing the most recent at the top now.

**Comment by kristinashu** [comment]
Confirmed, it's working correctly ( I swear it wasn't yesterday lol). 

image

**Comment by Pomax** [comment]
I think it shows the most recently edited page, rather than published - we can change that, but let's file a separate issue for that, as that's code in completely different files =P

**Comment by Pomax** [comment]
@kristinashu ready for re-review

**Comment by Pomax** [comment]
@mmmavis this is ready for code review

**Comment by Pomax** [comment]
Looks like we don't control that - I've filed https://github.com/wagtail/wagtail/issues/6080 so that the wagtail folks are aware of this issue.

**Pull request merged**